duckduckgo/iOS
# Rewrite Tracking Links (AMP Phase)

Task/Issue URL: https://app.asana.com/0/72649045549333/1201286795398788/f
Tech Design URL: https://app.asana.com/0/481882893211075/1201299313936485/f
CC: @bwaresiak 

**Description**:
This PR adds support for extracting the canonical URL from AMP links

**Steps to test this PR**:
1. Navigate to https://privacy-test-pages.glitch.me/privacy-protections/amp/
2. Open the links on the page. The loaded url should match the "Expected" url from the test page.
3. Paste https://www.bbc.com/news/world-asia-china-19432800.amp into the address bar. The loaded page should be https://www.bbc.com/news/world-asia-china-19432800
4. Paste https://www.bbc.com/news/world-asia-china-19432800.amp into another app such as notes or reminders. Open the link with DDG as the default browser. The loaded page should be https://www.bbc.com/news/world-asia-china-19432800



**Device Testing**:


**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
